### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by adrianmasiewicz**

### CODE REVIEW

File: module2/homework/generateSequence/generateSequence.hpp [hunk 1]
```diff
@@ -2,6 +2,9 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    std::vector<int> vec;
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:5]: You can pre-allocate all the required memory

File: module2/homework/nwdNww/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+        while (true)
+        {
+            while(lhs >= rhs){
+                if(lhs == rhs) return rhs;
+                lhs -= rhs;
+            }
+            int number = lhs;
+            lhs = rhs;
+            rhs = number;
+        }
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:14]: That 'if' requires brackets. 
I know that it works even without them, but it's ugly.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+    
+    return 0;
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+      if ((lhs > 0) && (rhs > 0))
+        return lhs * rhs / NWD(lhs, rhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:21]: I'm copying my comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.


The simplest change that i saw in other solutions was to simply change the order of execution, ie.:
return (lhs / nwd(lhs, rhs)) * rhs.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+
+    if (lhs < 0) lhs *= -1;
+    if (rhs < 0) rhs *= -1;
+
+    if(lhs==rhs && lhs == 0) return 0;
```
Comment (KrzysztofJ85) [inline module2/homework/nwdNww/nwdNww.hpp:24]: Try to be consistent with spaces :)

File: module2/homework/addEven/addEven.hpp [hunk 5]
```diff
@@ -2,7 +2,17 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int liczba; //liczba elementów w wektorze
+    int liczba3 = 0; //suma liczb zwracana przez funkcję
+    liczba = numbers.size();
```
Comment (KrzysztofJ85) [inline module2/homework/addEven/addEven.hpp:7]: It's good practice to use english words for variable names.

Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:7]: use only English number

File: module2/homework/addEven/addEven.hpp [hunk 6]
```diff
@@ -2,7 +2,17 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int liczba; //liczba elementów w wektorze
+    int liczba3 = 0; //suma liczb zwracana przez funkcję
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:6]: if you need comment what variable store it's mean you named it wrong

File: module2/homework/addEven/addEven.hpp [hunk 7]
```diff
@@ -2,7 +2,17 @@
...
+    liczba = numbers.size();
+
+    for (int liczba2 = 0 && liczba3; liczba2<liczba; liczba2++) {
+
+        if (numbers[liczba2]%2==0) {
+
+            liczba3=numbers[liczba2]+liczba3;
+        }
+    }
+
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:16]: wrong format

File: module2/homework/addEven/addEven.hpp [hunk 8]
```diff
@@ -2,7 +2,17 @@
...
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int liczba; //liczba elementów w wektorze
+    int liczba3 = 0; //suma liczb zwracana przez funkcję
+    liczba = numbers.size();
+
+    for (int liczba2 = 0 && liczba3; liczba2<liczba; liczba2++) {
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:9]: try simplyfy this foor loop

File: module2/homework/maxOfVector/main.cpp [hunk 9]
```diff
@@ -5,6 +5,6 @@
 int main() {
     std::vector<int> vec{4, 3, 2, 5, 6, 7, 9, 1, 0};
     std::cout << maxOfVector(vec) << '\n';
-
+    
     return 0;
 }
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/main.cpp:10]: you don't need to commit this file

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 10]
```diff
@@ -2,7 +2,19 @@
...
+    liczba = numbers.size();
+    liczba3 = numbers.front();
+
+    for (int liczba2 = 0 && liczba3; liczba2<liczba; liczba2++) {
+        if(liczba3<numbers[liczba2]){
+            liczba3 = numbers[liczba2];
+        }
+
+    }
+
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:18]: if you need comment what variable store it's mean you named it wrong

File: module2/homework/nwdNww/nwdNww.hpp (hunk 4 above)
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:24]: wrong format

File: module2/homework/addEven/addEven.hpp (hunk 7 above)
Comment (Marimeiko) [reply module2/homework/addEven/addEven.hpp:16]: > wrong format
What do you mean by that?

File: module2/homework/addEven/addEven.hpp (hunk 6 above)
Comment (ziobron) [reply module2/homework/addEven/addEven.hpp:6]: yes, please just name a variable in the way you wanted to comment it
`int numberOfVectorElements` is really descriptive.

File: module2/homework/addEven/addEven.hpp (hunk 7 above)
Comment (ziobron) [reply module2/homework/addEven/addEven.hpp:16]: He meant bad code formatting :) I suggest something like that:
```suggestion
    for (int liczba2 = 0 && liczba3; liczba2 < liczba; liczba2++) {
        if (numbers[liczba2] % 2 == 0) {
            liczba3 = numbers[liczba2] + liczba3;
        }
    }
```

Variable names also should be more descriptive.

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05
+3 XP cooperation

{
  "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635, 1636],
  "points": [2, 4, 2, 6, 2, 5, 2, 5, 3],
  "users": ["adrianmasiewicz", "Marimeiko"]
}

**Pull request closed without merging**